consensus/safety-rules/src/safety_rules.rs

```diff
@@ -249,26 +260,31 @@ impl SafetyRules {
 
         let author = self.persistent_storage.author()?;
         if let Some(expected_key) = epoch_state.verifier.get_public_key(&author) {
-            let curr_key = self.signer().ok().map(|s| s.public_key());
+            let curr_key = self.validator_handle().ok().map(|s| s.key_version());
             if curr_key != Some(expected_key.clone()) {
-                let consensus_key = self
-                    .persistent_storage
-                    .consensus_key_for_version(expected_key)
-                    .map_err(|e| {
-                        let error = Error::InternalError(format!(
-                            "Validator key not found: {:?}",
-                            e.to_string()
-                        ));
-                        error!(
-                            SafetyLogSchema::new(LogEntry::KeyReconciliation, LogEvent::Error)
-                                .error(&error),
-                        );
-
-                        self.validator_signer = None;
-                        error
-                    })?;
-
-                self.validator_signer = Some(ValidatorSigner::new(author, consensus_key));
+                let validator_handle = Some(ValidatorHandle::new(
+                    author,
+                    CONSENSUS_KEY.into(),
```

Comment:
this is a const, why are we introducing it as a field?

Reply:
Fair enough -- if we're always assuming a validator is identified by the CONSENSUS_KEY, I'll merge this into the ValidatorHandle 😄